health/health.go

```diff
@@ -3,26 +3,44 @@
 package health
 
 import (
-	"time"
+	"sync"
 
 	"golang.org/x/net/context"
 	"google.golang.org/grpc"
-	healthpb "google.golang.org/grpc/health/grpc_health"
+	"google.golang.org/grpc/codes"
+	healthpb "google.golang.org/grpc/health/grpc_health_v1alpha"
 )
 
-// HealthCheck is the client side function to health-check a server
-func HealthCheck(t time.Duration, cc *grpc.ClientConn) error {
-	ctx, _ := context.WithTimeout(context.Background(), t)
-	hc := healthpb.NewHealthCheckClient(cc)
-	req := new(healthpb.HealthCheckRequest)
-	_, err := hc.Check(ctx, req)
-	return err
+type HealthServer struct {
+	// StatusMap stores the serving status of a service
```

Comment:
statusMap stores the serving statuses of the services this HealthServer monitors.